rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -32,6 +32,7 @@
 #include "rclcpp/allocator/allocator_deleter.hpp"
 #include "rclcpp/detail/resolve_use_intra_process.hpp"
 #include "rclcpp/experimental/intra_process_manager.hpp"
+#include "rclcpp/generic_publisher.hpp"
```

Comment:
I am not sure if the two template functions are worth putting them into a separate file. Also, given that there are only functions, I don't think they should live in a `generic_publisher` namespace.